File: backend/src/controllers/health.ts
import { Router } from 'express';
import { sendResponse } from '../utils/response.js';

const router = Router();

/**
 * @swagger
 * /api/v1/health:
 *   get:
 *     summary: Server health check
 *     tags: [Health]
 *     responses:
 *       200:
 *         description: Server is running normally
 *         content:
 *           application/json:
 *             schema:
 *               $ref: '#/components/schemas/HealthResponse'
 */
router.get('/health', (_req, res) => {
  sendResponse(res, {
    message: 'Máy chủ hoạt động bình thường',
    message_en: 'Server is running well!',
    data: { timestamp: new Date().toISOString() },
  });
});

export default router;

File: backend/app/routers/health.py
from fastapi import APIRouter
from app.utils.response import send_response
from app.schemas.common import ApiResponse
from typing import Any

router = APIRouter()

@router.get("/health", response_model=ApiResponse[Any])
async def health_check():
    return send_response(data={"status": "ok"}, message="Server is healthy", message_en="Server is healthy")
